app/javascript/packs/application.js

```diff
@@ -20,6 +21,24 @@ global.mapsLoadedCallback = function() {
   }
 }
 
+Accordion.prototype.originalSetExpanded = Accordion.prototype.setExpanded;
+Accordion.prototype.setExpanded = function (expanded, $section) {
+  const eventName = this.$module.attributes['data-track-event-name'];
+
+  if (eventName) {
+    const segment = $section.querySelector('button').id;
+
+    if (expanded && !$section.classList.contains(this.sectionExpandedClass)) {
+      sendGAEvent(eventName.value, segment, 'open');
```

Comment:
I'm quite aware this moves the GA interaction code into the setExpanded function - I'd originally been thinking the monkey patch would trigger an event within the page.

This can be split this into 'emit a javascript event' monkey patch, then 'catch the javascript event, and interact with GA' as separate code.

This then makes the GA interaction asynchronous relative to the setExpanded function, and segregates the code.

eg

```javascript
if (expanded && !$section.classList.contains(this.sectionExpandedClass)) {
    let accordianExpanded = new CustomEvent('accordiantoggle', expanded: expanded);
    $section.dispatchEvent(accordianExpanded) ;
     
```

and then to catch the event

```javascript
document.addEventListener('accordianexpanded', function(e) {
  sendGAEvent(eventName.value, segment, e.expanded ? 'open' : 'closed');
}) ;
```

The above uses the new CustomEvent syntax - https://developer.mozilla.org/en-US/docs/Web/Guide/Events/Creating_and_triggering_events

You'll either need the old IE compatible syntax, or to add the polyfill
https://www.npmjs.com/package/custom-event-polyfill